src/Filesystem/Provider/FileStreamAccess.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace ILIAS\Filesystem\Provider;
+
+use ILIAS\Filesystem\Exception\FileAlreadyExistsException;
+use ILIAS\Filesystem\Exception\FileNotFoundException;
+use ILIAS\Filesystem\Exception\IOException;
+use Psr\Http\Message\StreamInterface;
```

Comment:
We had this discussion, too. But I think the public interface is more important than which libraries are used. And an own implementation of Streams would be a overkill, IMO.